src/kademlia/node.cpp

```diff
@@ -436,12 +435,12 @@ void node::get_item(sha1_hash const& target
 #endif
 
 	boost::intrusive_ptr<dht::get_item> ta;
-	ta.reset(new dht::get_item(*this, target, f));
+	ta.reset(new dht::get_item(*this, target, boost::bind(f, _1), find_data::nodes_callback()));
```

Comment:
like boost::shared_ptr ta = boost::make_shared<dht::get_item>(*this, target, f) ?

Usually i use std::make_shared (with std=c++11 for sure, and I know libtorrent also support c98 so we couldn't use it), I don't have any experience about boost::intrusive_ptr. But in kademlia, seems all codes are using it, not shared_ptr.